pkg/ingestor/puller/blob/blob_test.go

```diff
@@ -0,0 +1,525 @@
+				},
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_21j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster1.k8s.local/",
+				},
+				{
+					Key: "cluster2.k8s.local/kubehound_cluster2.k8s.local_01j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster2.k8s.local/kubehound_cluster2.k8s.local_11j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster2.k8s.local/",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Sanitize path",
+			fields: fields{
+				bucketName: bucketName,
+			},
+			args: args{
+				recursive: true,
+				prefix:    "cluster1.k8s.local",
+			},
+			want: []*puller.ListObject{
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_01j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_11j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_21j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster1.k8s.local/",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Sanitize path",
+			fields: fields{
+				bucketName: bucketName,
+			},
+			args: args{
+				recursive: true,
+				prefix:    "cluster1.k8s.local/",
+			},
+			want: []*puller.ListObject{
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_01j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_11j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+				{
+					Key: "cluster1.k8s.local/kubehound_cluster1.k8s.local_21j2qs8th6yarr5hkafysekn0j.tar.gz",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Sanitize path",
+			fields: fields{
+				bucketName: bucketName,
+			},
+			args: args{
+				recursive: false,
+				prefix:    "",
+			},
+			want: []*puller.ListObject{
+				{
+					Key: "cluster1.k8s.local/",
+				},
+				{
+					Key: "cluster2.k8s.local/",
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctx := context.Background()
+			bs := &BlobStore{
+				bucketName: tt.fields.bucketName,
+				cfg:        tt.fields.cfg,
+				region:     tt.fields.region,
+			}
+			got, err := bs.ListFiles(ctx, tt.args.prefix, tt.args.recursive)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("BlobStore.ListFiles() error = %v, wantErr %v", err, tt.wantErr)
+
+				return
+			}
+
+			// Reset modtime to avoid comparison issues
+			for _, v := range got {
+				v.ModTime = time.Time{}
+			}
+
+			if !reflect.DeepEqual(got, tt.want) {
+				for i, v := range got {
+					t.Logf("Got: %d: %s", i, v.Key)
+				}
+				t.Errorf("BlobStore.ListFiles() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBlobStore_Pull(t *testing.T) {
+	t.Parallel()
+
+	// Get current working directory to pass SaneCheckPath()
+	pwd, err := os.Getwd()
```

Comment:
Is there really no workaround? having a test depending on the path where it is launched sounds like a bad idea